dist/tools/bossa/bossa.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/sh
+#
+# Unified BOSSA script for RIOT
+#
+# This script is supposed to be called from RIOTs make system,
+# as it depends on certain environment variables:
+#
+# BOSSAC:       path to `bossac` binary
+# HEXFILE:      path to compiled (RIOT) binary, despite the name it must be in
+#               binary format (not ihex or similar)
+# PORT:         serial port connected to the board, e.g. `/dev/ttyACM0`
+#
+# [optional options:]
+# BOSSAC_FLAGS: flags used when calling the bossac command, default is "-i -e -w -v -b -R"
+# BOSSA_ARDUINO_PREPARE:
+#               this command will configure the given PORT to 1200 baud, needed
+#               to flash certain Arduino boards (e.g. arduino-due)
+#
+# @author       Hauke Petersen <hauke.petersen@fu-berlin.de>
+#
+
+# first lets check some parameters (specified by command line vars)
+if [ -z "${BOSSAC}" ]; then
+    echo "error: bossac binary not specified"
+    exit 1
+fi
+if [ -z "${BOSSAC_FLAGS}" ]; then
+    BOSSAC_FLAGS="-i -e -w -v -b -R"
+fi
+if [ -z "${HEXFILE}" ]; then
+    echo "error: path to RIOT binary not specified"
+    exit 1
+fi
+if [ -z "${PORT}" ]; then
+    echo "error: no PORT specified"
+    exit 1
+fi
+
+if [ `uname` = "Linux" ]; then
+    STTY_FLAG=-F
+elif [ `uname` = "Darwin" ]; then
+    STTY_FLAG=-f
+else
+    echo "error: your host system is not support by the bossa.sh script"
+    exit 1
+fi
+
+# check if we need to put an Arduino into flash mode
+if [ ! -z "${BOSSA_ARDUINO_PREPARE}" ]; then
+    stty "${STTY_FLAG}" "${PORT}" raw ispeed 1200 ospeed 1200 cs8 -cstopb ignpar eol 255 eof 255
+fi
+
+# now we flash the binary onto the target board
+${BOSSAC} --port="${PORT}" ${BOSSAC_FLAGS} "${HEXFILE}"
```

Comment:
Please change the script to not use environment but cmdline for options (as far as easily possible).

Setting 

```
FFLAGS:="-i -e -w -v -b -R --PORT=$(PORT) $(HEXFILE)"
```
in makefiles/tools/bossac.mk and using

```
${BOSSAC} $*
```

here keeps the configuration whithin the makefiles, and more consistent for different flashers.

Ideally you'd also create a phony target for stty, depend on that for flashing, and drop the shell script altogether.